ui/src/timeMachine/components/TimeMachineFluxEditor.tsx

```diff
@@ -56,31 +56,58 @@ const TimeMachineFluxEditor: FC<Props> = ({
     onSetActiveQueryText(editorInstance.getValue())
   }
 
+  const getInsertLineNumber = (
+    currentLineNumber: number,
+    scriptLines: string[]
+  ): number => {
+    const currentLine =
+      scriptLines[currentLineNumber] || scriptLines[scriptLines.length - 1]
+
+    // Insert on the current line if its an empty line
+    if (!currentLine.trim()) {
+      return currentLineNumber
+    }
+
+    return currentLineNumber + 1
+  }
+
   const handleInsertFluxFunction = (func: FluxToolbarFunction): void => {
     const p = editorInstance.getPosition()
+    const scriptLines = activeQueryText.split('\n')
+
+    const insertLineNumber = getInsertLineNumber(p.lineNumber, scriptLines)
+
     // sets the range based on the current position
     let range = new window.monaco.Range(
-      p.lineNumber,
-      p.column,
-      p.lineNumber,
-      p.column
+      insertLineNumber, // the row beneath the cursor
+      1, // beginning column of the row
+      insertLineNumber,
+      1
```

Comment:
can you add a comment here explaining what this magic number is, or store it into a variable that's named what `1` represents?

Reply:
I think it might be a little redundant. The gist of the `Range` monaco object is to define the Range of the editor:

https://microsoft.github.io/monaco-editor/api/classes/monaco.range.html

In this case the `1` represents the start and end position of the column (i.e. the beginning of the column). We can add another comment, but it might be excessive don't you think?